jacsstorage-services/src/main/java/org/janelia/jacsstorage/service/s3/S3AdapterProvider.java

```diff
@@ -0,0 +1,17 @@
+package org.janelia.jacsstorage.service.s3;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import javax.inject.Singleton;
+
+import org.janelia.jacsstorage.model.jacsstorage.JADEOptions;
+
+@Singleton
+public class S3AdapterProvider {
+    private static final ConcurrentMap<String, S3Adapter> S3_ADAPTERS = new ConcurrentHashMap<>();
+
+    public S3Adapter getS3Adapter(String bucket, String endpoint, JADEOptions s3Options) {
+        return S3_ADAPTERS.computeIfAbsent(bucket, (b) -> new S3Adapter(b, endpoint, s3Options));
```

Comment:
If you have a bucket with the same name in both AWS and VAST, this would only cache one of them. Why not key on endpoint+bucket?

Reply:
that's a good point - will do it